File: app/src/main/java/com/example/zhpan/circleviewpager/activity/BaseDataActivity.java
package com.example.zhpan.circleviewpager.activity;


import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

public abstract class BaseDataActivity extends AppCompatActivity {

    protected List<Integer> mDrawableList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initData();
    }

    private void initData() {
        for (int i = 0; i <= 2; i++) {
            int drawable = getResources().getIdentifier("guide" + i, "drawable", getPackageName());
            mDrawableList.add(drawable);
        }
    }
}

File: app/src/main/java/com/example/zhpan/circleviewpager/activity/BaseDataActivity.kt
package com.example.zhpan.circleviewpager.activity


import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import java.util.ArrayList

abstract class BaseDataActivity : AppCompatActivity() {

    protected var mDrawableList: MutableList<Int> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initData()
    }

    private fun initData() {
        for (i in 0..2) {
            val drawable = resources.getIdentifier("guide$i", "drawable", packageName)
            mDrawableList.add(drawable)
        }
    }
}
